src/web/js/trove/chart-lib.js

```diff
@@ -1207,36 +1152,66 @@
+      data.addColumn({id: 'i1', type: 'number', role: 'interval'});
     });
 
+    // console.log('legends is', legends);
+    // console.log('data I is', data);
+
     combined.forEach((p, i) => {
-      /*
-      x | n n n | y | n n n n n n n n n n n n
-            i         combined.length - i - 1
-      */
-      const prefix = new Array(2 * i).fill(null);
-      const suffix = new Array(2 * (combined.length - i - 1)).fill(null);
-      const rowTemplate = [0].concat(prefix).concat([null, null]).concat(suffix);
-      data.addRows(get(p, 'ps').map(row => {
-        const currentRow = rowTemplate.slice();
-        if (row.length != 0) {
-          currentRow[0] = toFixnum(row[0]);
-          currentRow[2*i + 1] = toFixnum(row[1]);
-          let labelRow = null;
-          if (row.length >= 3 && row[2] !== '') {
-            labelRow = `<p>label: <b>${row[2]}</b></p>`;
-          } else {
-            labelRow = '';
+      // console.log('Ncols is', combined.length * 4);
+      const rowTemplate = new Array(combined.length * 4 + 1).fill(null);
+      const intervalP = (i >= minIntervalIndex);
+      // console.log('rowTemplate is', rowTemplate);
+      if (!intervalP) {
+        data.addRows(get(p, 'ps').map(row => {
+          // console.log('this row is', row);
+          const currentRow = rowTemplate.slice();
+          if (row.length != 0) {
+            currentRow[0] = toFixnum(row[0]);
+            currentRow[4*i + 1] = toFixnum(row[1]);
+            let labelRow = null;
+            if (row.length >= 3 && row[2] !== '') {
+              labelRow = `<p>label: <b>${row[2]}</b></p>`;
+            } else {
+              labelRow = '';
+            }
+            currentRow[4*i + 2] = `<p>${legends[i]}</p>
+<p>x: <b>${currentRow[0]}</b></p>
+<p>y: <b>${currentRow[4*i + 1]}</b></p>
+              ${labelRow}`;
+            // currentRow[4*i + 3] = 0;
+            // currentRow[4*i + 4] = 0;
           }
-          currentRow[2*i + 2] = `<p>${legends[i]}</p>
+          return currentRow;
+        }));
+      } else {
+        data.addRows(get(p, 'tab').map(row => {
+          const currentRow = rowTemplate.slice();
+          if (row.length != 0) {
+            currentRow[0] = toFixnum(row[0]);
+            currentRow[4*i + 1] = toFixnum(row[1]);
+            let labelRow = null;
+            if (row.length >= 3 && row[2] !== '') {
+              labelRow = `<p>label: <b>${row[2]}</b></p>`;
+            } else {
+              labelRow = '';
+            }
+            currentRow[4*i + 2] = `<p>${legends[i]}</p>
 <p>x: <b>${currentRow[0]}</b></p>
-<p>y: <b>${currentRow[2*i + 1]}</b></p>
-${labelRow}`;
-        }
-        return currentRow;
-      }));
+<p>y: <b>${currentRow[4*i + 1]}</b></p>
+              ${labelRow}`;
+            currentRow[4*i + 3] = row[1];
+            currentRow[4*i + 4] = row[2];
+          }
+          return currentRow;
+        }));
+      };
     });
 
+    // console.log('row setting done');
```

Comment:
Lots of commented out `console.log`s – typically our style is to remove these though if there's a good reason for leaving them in because they were difficult to design could leave a comment to that effect.